src/Compilers/CSharp/Portable/Compiler/ClsComplianceChecker.cs

```diff
@@ -52,7 +57,13 @@ public static void CheckCompliance(CSharpCompilation compilation, DiagnosticBag
         {
             var queue = new ConcurrentQueue<Diagnostic>();
             var checker = new ClsComplianceChecker(compilation, filterTree, filterSpanWithinTree, queue, cancellationToken);
+            if (compilation.Options.ConcurrentBuild)
```

Comment:
why not check the option in side of constructor of ClsCompilanceChecker? seems it is okay to create the stack in the ctor with readonly?

Reply:
I was following the pattern of `MethodCompiler`. I see that the VB implementation of the same feature initializes the collection in the constructor. I'll move it there for the C# one.